### DISCUSSION TITLE
Using `:pinAllExceptPeerDependencies` for some packages and `:pinOnlyDevDependencies` for others - correct config?

### DISCUSSION THREAD

**Discussion by darrellwarde** [renovatebot/renovate#17344] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

About a week ago, Renovate suddenly pinned all of our dependencies and peer dependencies, despite using the `config:js-lib` preset. You can see the PR where this happened here: https://github.com/neo4j/graphql/pull/1914

There were no changes to the config around that time, so it is unclear what triggered this.

We publish this repository as a library, and we have since published a broken release without realising this had changed.

### Relevant debug logs

<details><summary>Logs</summary>

Logs for this run a week ago to not appear to be available on the dashboard.

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction, but I have linked to a public repo where it occurs

**Comment by rarkins** [comment]
The behavior change occurred immediately after this config change: https://github.com/neo4j/graphql/commit/4b874e922bf2e20638860c044928a6f1c40106d3

You can see there's a reference to `config:js-app` for a single package there. These `config:*` presets are defined as "[Full config presets](https://docs.renovatebot.com/presets-config/)" and not intended to be mixed with packageRules like that.

Here is the preset which gets pulled in: https://docs.renovatebot.com/presets-default/#pinallexceptpeerdependencies

**Comment by darrellwarde** [comment]
I see, I was under the impression that this was valid from the documentation, with the fourth code block under https://docs.renovatebot.com/configuration-options/#packagerules demonstrating this - but I assume this works because this is not a `config:*` scoped preset? Thank you for the guidance, I will get this fixed!

**Comment by rarkins** [comment]
Yes. It's not that we "want" this type of behaviour but the config preset included another preset which overrode all except peer dependencies to pin.

**Comment by darrellwarde** [comment]
Sure, I get that, thank you for the help, I'll close this issue. 🙂

**Comment by rarkins** [comment]
Sorry for the trouble it caused you and your users! Feel free to create a fresh discussion in this repo in future if you'd like any new config reviewed which you're unsure of

**Comment by darrellwarde** [comment]
Perhaps this is now unmonitored as a closed issue, but could something along the lines of the following config help remedy this situation? `@neo4j/graphql-toolbox` is our only `js-app` so essentially try to reproduce the `config:js-app` preset for that, and the `config:js-lib` preset for all others. Thank you for the quick help today - we love Renovate and all of the time it saves us! 🙏 

```json
{
    "extends": ["config:base"],
    ...
    "packageRules": [
        {
            "matchPackageNames": ["@neo4j/graphql-toolbox"],
            "extends": [":pinAllExceptPeerDependencies"]
        },
        {
            "matchPackagePatterns": ["*"],
            "excludePackageNames": ["@neo4j/graphql-toolbox"],
            "extends": [":pinOnlyDevDependencies"]
        },
    ...
```
